### PR TITLE
Add requests API

### PR DESCRIPTION
**Pull request by niknetniko**
Only index and confirm/decline/cancel. ~~No clue if this introduces Big Security :tm: holes.~~

### CODE REVIEW

**Comment by niknetniko** [comment]
There are now tests, so no more bugs :)

File: app/controllers/api/v1/requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+module Api
+  module V1
+    class RequestsController < Api::V1::ApplicationController
+      load_and_authorize_resource :user, find_by: :name
+      before_action :load_request, only: [:confirm, :decline, :cancel]
+      authorize_resource :request, id_param: :request_id, only: [:confirm, :decline, :cancel]
+
+      def index
+        all_requests = @current_user.requests.includes(:debtor, :creditor, :issuer).group_by(&:status)
```
Comment (TomNaessens) [inline app/controllers/api/v1/requests_controller.rb:11]: Does current_user need an @?

File: app/controllers/api/v1/requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              actions: []
+            }
+            mapped[:actions] << :confirm if can?(:confirm, t)
+            mapped[:actions] << :cancel if can?(:cancel, t)
+            mapped[:actions] << :decline if can?(:decline, t)
+            mapped
+          end
+        end
+
+        render json: mapped_requests
```
Comment (TomNaessens) [inline app/controllers/api/v1/requests_controller.rb:32]: Could be a jbuilder so the representation of the request is extracted from the controller code

**Comment by niknetniko** [comment]
I have redone this:

- Changed the API to be one list, with optional filtering on status.
- Use rswag tests

**Review by redfast00 — APPROVED** [review]

**Pull request merged**